packages/cli/src/commands/test.ts

```diff
@@ -164,45 +163,51 @@ export default class Test extends Command {
     await runner.run()
   }
 
-  // eslint-disable-next-line require-await
   async prepareRunLocation (
     configOptions: { runLocation?: string, privateRunLocation?: string } = {},
     cliFlags: { runLocation?: string, privateRunLocation?: string },
   ): Promise<RunLocation> {
     // Command line options take precedence
     if (cliFlags.runLocation) {
-      return { type: 'PUBLIC', region: cliFlags.runLocation }
+      const { data: availableLocations } = await api.locations.getAll()
+      if (availableLocations.some(l => l.region === cliFlags.runLocation)) {
+        return Promise.resolve({ type: 'PUBLIC', region: cliFlags.runLocation })
+      }
+      throw new Error(`Unable to run checks on unsupported location "${cliFlags.runLocation}". ` +
+        `Supported locations are:\n${availableLocations.map(l => `${l.region}`).join('\n')}`)
     } else if (cliFlags.privateRunLocation) {
       return this.preparePrivateRunLocation(cliFlags.privateRunLocation)
     } else if (configOptions.runLocation && configOptions.privateRunLocation) {
-      this.error(
-        'Both runLocation and privateRunLocation fields were set in the Checkly config file.' +
-        ` Please only specify one run location. The configured locations were "${configOptions.runLocation}" and "${configOptions.privateRunLocation}"`,
-        { exit: 1 },
-      )
+      throw new Error('Both runLocation and privateRunLocation fields were set in the Checkly config file.' +
+        ` Please only specify one run location. The configured locations were' + 
+        ' "${configOptions.runLocation}" and "${configOptions.privateRunLocation}"`)
     } else if (configOptions.runLocation) {
-      return { type: 'PUBLIC', region: configOptions.runLocation }
+      return Promise.resolve({ type: 'PUBLIC', region: configOptions.runLocation })
     } else if (configOptions.privateRunLocation) {
       return this.preparePrivateRunLocation(configOptions.privateRunLocation)
     } else {
-      return { type: 'PUBLIC', region: DEFAULT_REGION }
+      return Promise.resolve({ type: 'PUBLIC', region: DEFAULT_REGION })
     }
   }
 
   async preparePrivateRunLocation (privateLocationSlugName: string): Promise<PrivateRunLocation> {
-    const { data: privateLocations } = await api.privateLocations.getAll()
-    const privateLocation = privateLocations.find(({ slugName }) => slugName === privateLocationSlugName)
-    if (privateLocation) {
-      return { type: 'PRIVATE', id: privateLocation.id, slugName: privateLocationSlugName }
+    try {
+      const { data: privateLocations } = await api.privateLocations.getAll()
+      const privateLocation = privateLocations.find(({ slugName }) => slugName === privateLocationSlugName)
+      if (privateLocation) {
+        return { type: 'PRIVATE', id: privateLocation.id, slugName: privateLocationSlugName }
+      }
+      // We can use a null-assertion operator safely since account ID was validated in auth-check hook
```

Comment:
This comment isn't needed anymore now that the `!` operator is gone from `config.getAccountId()!`